### PR TITLE
feat(labs): use redux 

### PR DESCRIPTION
**Pull request by jackcmeyer**
Adds redux for requesting, canceling, completing, and updating a lab

### CODE REVIEW

File: src/__tests__/labs/lab-slice.test.ts [hunk 1]
```diff
@@ -0,0 +1,374 @@
+import thunk from 'redux-thunk'
+import createMockStore from 'redux-mock-store'
+import PatientRepository from '../../clients/db/PatientRepository'
+import LabRepository from '../../clients/db/LabRepository'
+import Lab from '../../model/Lab'
+import Patient from '../../model/Patient'
+import * as labSlice from '../../labs/lab-slice'
+import { RootState } from '../../store'
+import { requestLab } from '../../labs/lab-slice'
```
Comment (fox1t) [inline src/__tests__/labs/lab-slice.test.ts:9]: Join this import with the line 7

File: src/__tests__/labs/lab-slice.test.ts [hunk 2]
```diff
@@ -0,0 +1,374 @@
+import thunk from 'redux-thunk'
+import createMockStore from 'redux-mock-store'
+import PatientRepository from '../../clients/db/PatientRepository'
+import LabRepository from '../../clients/db/LabRepository'
+import Lab from '../../model/Lab'
+import Patient from '../../model/Patient'
+import * as labSlice from '../../labs/lab-slice'
```
Comment (fox1t) [inline src/__tests__/labs/lab-slice.test.ts:7]: Remove * as

File: src/__tests__/labs/requests/NewLabRequest.test.tsx [hunk 3]
```diff
@@ -48,7 +48,7 @@ describe('New Lab Request', () => {
     const history = createMemoryHistory()
 
     beforeEach(() => {
-      const store = mockStore({ title: '' })
+      const store = mockStore({ title: '', lab: { status: 'loading', error: {} } })
```
Comment (fox1t) [inline src/__tests__/labs/requests/NewLabRequest.test.tsx:51]: I really like explicit status in place of isLoading boolean. We have to use this approach in more places.

**Comment by fox1t** [comment]
LGTM

**Comment by jackcmeyer** [comment]
@fox1t I have addressed all of your comments :)

**Pull request merged**